test/modules/GenericMqttTester/src/message_channel.rs

```diff
@@ -0,0 +1,160 @@
+use async_trait::async_trait;
+use futures_util::{
+    future::{self, Either},
+    stream::StreamExt,
+};
+use mpsc::{Receiver, Sender, UnboundedReceiver, UnboundedSender};
+use tokio::sync::mpsc;
+use tracing::{error, info};
+
+use mqtt3::{
+    proto::{Publication, QoS},
+    PublishHandle, ReceivedPublication,
+};
+
+use crate::{MessageTesterError, BACKWARDS_TOPIC};
+
+#[derive(Debug, Clone)]
+pub struct MessageHandlerShutdownHandle(Sender<()>);
```

Comment:
nit: just `ShutdownHandle`?

Reply:
I was trying to differentiate between the MessageTesterShuttdownHandle and this one. Now that you mention it, I feel like it should be named MessageChannelShutdownHandle.